src/python/WMCore/ReqMgr/Service/Request.py

```diff
@@ -406,24 +407,42 @@ def _handleNoStatusUpdate(self, workload, request_args, dn):
          1. RequestPriority
          2. Global workqueue statistics, while acquiring a workflow
         """
-        if 'RequestPriority' in request_args:
-            # Yes, we completely ignore any other arguments posted by the user (web UI case)
-            request_args = {'RequestPriority': request_args['RequestPriority']}
-            validate_request_priority(request_args)
-            # must update three places: GQ elements, workload_cache and workload spec
-            self.gq_service.updatePriority(workload.name(), request_args['RequestPriority'])
-            report = self.reqmgr_db_service.updateRequestProperty(workload.name(), request_args, dn)
-            workload.setPriority(request_args['RequestPriority'])
-            workload.saveCouchUrl(workload.specUrl())
-            cherrypy.log('Updated priority of "{}" to: {}'.format(workload.name(), request_args['RequestPriority']))
-        elif workqueue_stat_validation(request_args):
-            report = self.reqmgr_db_service.updateRequestStats(workload.name(), request_args)
-            cherrypy.log('Updated workqueue statistics of "{}", with:  {}'.format(workload.name(), request_args))
-        else:
-            msg = "There are invalid arguments for no-status update: %s" % request_args
+        report = []
+        reqArgsDiff = deepcopy(request_args)
+
+        if not reqArgsDiff:
+            cherrypy.log("Nothing to be changed at this stage")
+            return reduceReport(report)
+
+        reqArgsNothandled = []
+        for reqArg in reqArgsDiff.copy():
+            if reqArg == 'RequestPriority':
+                validate_request_priority(reqArgsDiff)
+                # must update three places: GQ elements, workload_cache and workload spec
+                self.gq_service.updatePriority(workload.name(), reqArgsDiff['RequestPriority'])
+                workload.setPriority(reqArgsDiff['RequestPriority'])
+                workload.saveCouchUrl(workload.specUrl())
```

Comment:
For this call:
```
workload.saveCouchUrl(workload.specUrl())
```

I think it also needs to be executed for site list changes, such that those information get persisted in the workload object. Said that, I would suggest moving it outside of this loop, such that we only call it once.

Reply:
agreed, as we also  discussed it earlier during the chat

Reply:
done